IQSS/dataverse-frontend
# Feature/492 publish dataset with compare versions

## What this PR does / why we need it:
This PR adds a use case for comparing dataset versions, and uses it to add a property to the Dataset object, a boolean that indicates whether the next version should be a major or minor version update.
The boolean requiresMajorVersionUpdate will be true if the draft contains added, removed or replaced files.  If this boolean is true, then the PublishDatasetModal will include help text describing the major version update.

## Which issue(s) this PR closes:

- Closes #492 

## Special notes for your reviewer:

## Suggestions on how to test this:

1. Create and publish a dataset.
2. Edit the metadata of the dataset, and click the Publish button.  The PublishDatasetModal will show an option for choosing either minor or major version update.
3. Publish the dataset, and then create another draft version, with an uploaded, removed or replaced file.
4. Click the publish button again, and the PublishDatasetModal should display help text about the required major version update.


## Does this PR introduce a user interface change? If mockups are available, please link/include them here:

## Is there a release notes update needed for this change?:

## Additional documentation:
